### PR TITLE
Add PostgreSQL Superuser Permissions as a Prerequisite to Software Installation Docs

### LINKED ISSUE #1923: Add note about requiring Postgres superuser in software 0.30+
**Affected Document:**

All software install docs.

**Problem Type (Priority):**

- [ ] Error (Critical): This document has existing information that's incorrect and prevents readers from completing the documented setup (Add the `error/critical` label to this issue)
- [x] Misleading (Important): This document could benefit from additional context, guidance, or clarity (Add the `error/FYI` label to this issue)
- [ ] Typo (Low): Any issue related to spelling/ punctuation/ formatting (Add the `error/typo` label to this issue)

**Suggested Change (include links to relevant existing docs, support tickets, messages, or external content):**

**Additional Notes:**

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #1923 

image

### CODE REVIEW

**Review by jwitz — APPROVED** [review]

**Review by dintorf — CHANGES_REQUESTED** [review]
Looks good. I'd also make an edit to the [upgrade notes for version 0.30](https://github.com/astronomer/docs/blob/superuser/software_versioned_docs/version-0.30/upgrade-astronomer.md?plain=1#L205-L209). Something along these lines:

```markdown
#### Upgrading to 0.30+ PostgreSQL considerations

A change in 0.30 enabled the `trgm` extension for PosgreSQL. The schema migration that adds the extension requires a database superuser in order to execute. You can alter an existing user to be superuser with this SQL command: `ALTER USER <username> WITH SUPERUSER;`

If you use Azure Database for PostgreSQL as your database backend, you need to enable the `pg_trgm` extension before upgrading to Software 0.30 using either Azure portal or the Azure CLI. See [Azure documentation](https://docs.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-extensions) for configuration steps.

If you don't complete this setup before your upgrade, the upgrade will fail.
```

**Comment by kwcanuck** [comment]
@dintorf thanks for the quick review and feedback. Much appreciated.

I've modified the text as you recommended. Please review and provide your edits or approval.

To add edits directly to a PR:

image

File: software/upgrade-astronomer.md [hunk 1]
```diff
@@ -223,9 +223,14 @@ If you do specify the `--no-hooks` flag, the upgrade script will return a succes
 
 #### Upgrading to 0.30 when using Azure Database for PostgreSQL
```
Comment (dintorf) [inline software/upgrade-astronomer.md:224]: We need to change this header to something that states any postgres and not just Azure Database, like `Upgrading to 0.30 PostgreSQL requirements`

File: software_versioned_docs/version-0.30/upgrade-astronomer.md [hunk 2]
```diff
@@ -196,17 +196,24 @@ If you're upgrading to Astronomer Software 0.29 or later and Kubernetes 1.22 at
...
+Review tHe following considerations before upgrading to Astronomer Software 0.30.
+
 #### Running the 0.30 upgrade script with --no-hooks
 
 Using the `--no-hooks` flag in [Step 7](#step-7-run-astronomers-upgrade-script) results in the upgrade script skipping a necessary database migration job. Because of this, you should not specify this flag when running the upgrade script.
 
 If you do specify the `--no-hooks` flag, the upgrade script will return a success message even though it failed, resulting in broken behavior in your upgraded environment.
 
-#### Upgrading to 0.30 when using Azure Database for PostgreSQL
+#### Azure Database for PostgreSQL
```
Comment (dintorf) [inline software_versioned_docs/version-0.30/upgrade-astronomer.md:207]: We need to change this header to something that states any postgres and not just Azure Database, like `Upgrading to 0.30 PostgreSQL requirements`

**Review by dintorf — CHANGES_REQUESTED** [review]

File: software_versioned_docs/version-0.30/upgrade-astronomer.md (hunk 2 above)
Comment (kwcanuck) [inline software_versioned_docs/version-0.30/upgrade-astronomer.md:207]:
```suggestion
#### PostgreSQL requirements for Astronomer Software 0.30
```

File: software/upgrade-astronomer.md (hunk 1 above)
Comment (kwcanuck) [inline software/upgrade-astronomer.md:224]:
```suggestion
#### PostgreSQL requirements for Astronomer Software 0.30
```

**Review by dintorf — APPROVED** [review]
LGTM! thank you 👍

**Pull request merged**